synapse/federation/sender/__init__.py

```diff
@@ -201,7 +201,15 @@ async def handle_event(event: EventBase) -> None:
 
                     logger.debug("Sending %s to %r", event, destinations)
 
-                    self._send_pdu(event, destinations)
+                    if destinations:
+                        self._send_pdu(event, destinations)
+
+                        now = self.clock.time_msec()
+                        ts = await self.store.get_received_ts(event.event_id)
```

Comment:
Is there any concern about how much latency this will add by hitting the database again?

Reply:
Yes, though I think in this case its a *very* simple query and will be dwarfed by the complexity of the `get_hosts_in_room` query above

Reply:
Alright. 👍